File: LR2.cpp
//Мультисписок(вложенный список)

#include <iostream>
#include <vector>
int main() {
    std::vector<std::vector<int>> multilist = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };
}

//Очередь

#include <iostream>
#include <queue>
int main() {
    std::queue<int> q;
    q.push(1);
    q.push(2);
}

//Дек

#include <iostream>
#include <deque>
int main() {
    std::deque<int> deq;
    deq.push_back(1);  
    deq.push_front(2); 
    deq.push_back(3);
}

//Приоритетная очередь с примитивным типом

#include <iostream>
#include <queue>
int main() {
    std::priority_queue<int> pq;
    pq.push(10); 
    pq.push(20); 
    pq.push(5);  
}

//Приоритетная очередь с пользовательской структурой

#include <iostream>
#include <queue>
#include <string>
struct Task {
 std::string name;
 int priority;
 bool operator<(const Task& other) const {
 return priority < other.priority;
 }
};
int main() {
 std::priority_queue<Task> taskQueue;
 taskQueue.push({"Task 1", 2});
 taskQueue.push({"Task 2", 1});
 taskQueue.push({"Task 3", 3});
}

File: LR2.java
//Мультисписок(вложенный список)

import java.util.ArrayList;
import java.util.List;
public class Main {
    public static void main(String[] args) {
        List<List<Integer>> multilist = new ArrayList<>();
        multilist.add(List.of(1, 2, 3));
        multilist.add(List.of(4, 5, 6));
        multilist.add(List.of(7, 8, 9));
    }
}

//Очередь

import java.util.LinkedList;
import java.util.Queue;
public class QueueExample {
    public static void main(String[] args) {
        Queue<String> queue = new LinkedList<>();
        queue.offer("A"); 
        queue.offer("B");  
        queue.offer("C");
    }
}

//Дек

import java.util.Deque;
import java.util.ArrayDeque;
public class DequeExample {
    public static void main(String[] args) {
        Deque<Integer> deq = new ArrayDeque<>();
        deq.offerLast(1);  
        deq.offerFirst(2); 
        deq.offerLast(3); 
    }
}

//Приоритетная очередь

import java.util.PriorityQueue;
public class PriorityQueueExample {
    public static void main(String[] args) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        pq.offer(10);
        pq.offer(20);
        pq.offer(5);
    }
}

//Приоритетная очередь с компаратором:

Comparator<Task> idComparator = Comparator.comparing(Task::id);
PriorityQueue<Task> priorityQueue = new PriorityQueue<>(idComparator);
priorityQueue.add(new Task(10001, "Task 1", 5));
priorityQueue.add(new Task(10003, "Task 3", 10));
priorityQueue.add(new Task(10002, "Task 2", 1));
